pkg/kubectl/cmd/drain.go

```diff
@@ -657,3 +676,47 @@ func changed(desired bool) string {
 	}
 	return "uncordoned"
 }
+
+func (o *DrainOptions) sendEventWithMessage(desired bool) error {
+	reason := "NodeUncordoned"
+	if desired {
+		reason = "NodeCordoned"
+	}
+
+	t := metav1.Time{Time: clock.RealClock{}.Now()}
+	namespace := o.nodeInfo.Namespace
+	if namespace == "" {
+		namespace = metav1.NamespaceDefault
+	}
+
+	ref, err := api.GetReference(api.Scheme, o.nodeInfo.Object)
+
+	if err != nil {
+		return err
+	}
+
+	event := &api.Event{
```

Comment:
This highlights a need to make the codepath more reasonable, but at least this code is predictable now.